app/uk/gov/hmrc/economiccrimelevyreturns/controllers/StartController.scala

```diff
@@ -35,98 +39,122 @@ class StartController @Inject() (
   val controllerComponents: MessagesControllerComponents,
   authorise: AuthorisedAction,
   enrolmentStoreProxyService: EnrolmentStoreProxyService,
-  eclAccountConnector: EclAccountConnector,
-  eclReturnsService: EclReturnsService,
-  eclReturnsConnector: EclReturnsConnector,
+  eclAccountService: EclAccountService,
+  returnsService: ReturnsService,
   alreadySubmittedReturnView: AlreadySubmittedReturnView,
   noObligationForPeriodView: NoObligationForPeriodView,
   chooseReturnPeriodView: ChooseReturnPeriodView,
   view: StartView
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
-    with I18nSupport {
+    with I18nSupport
+    with BaseController
+    with ErrorHandler {
 
   def start(): Action[AnyContent] = authorise.async { implicit request =>
-    eclReturnsService.getOrCreateReturn(request.internalId).map { eclReturn =>
-      eclReturn.obligationDetails match {
-        case Some(obligationDetails) => Redirect(routes.StartController.onPageLoad(obligationDetails.periodKey))
-        case None                    => Ok(chooseReturnPeriodView())
-      }
-    }
+    implicit val hc: HeaderCarrier = CorrelationIdHelper.getOrCreateCorrelationId(request)
+    (for {
+      eclReturn <- returnsService.getOrCreateReturn(request.internalId).asResponseError
+    } yield eclReturn).fold(
+      error => Status(error.code.statusCode)(Json.toJson(error)),
```

Comment:
How does this error message manifest? We should be showing the default error page, but this looks like it'll show JSON like what's happening in Account server.